packages/mui-material/src/themeCssVarsAugmentation/index.d.ts

```diff
@@ -0,0 +1,60 @@
+import type {
+  CssVarsTheme,
+  ColorSystem,
+  PaletteCommonChannel,
+  PaletteColorChannel,
+  PaletteTextChannel,
+  PaletteActionChannel,
+  PaletteAppBar,
+  PaletteAvatar,
+  PaletteChip,
+  PaletteFilledInput,
+  PaletteLinearProgress,
+  PaletteSlider,
+  PaletteSnackbarContent,
+  PaletteStepConnector,
+  PaletteStepContent,
+  PaletteSwitch,
+  PaletteTableCell,
+  PaletteTooltip,
+} from '../styles/experimental_extendTheme';
+
+/**
+ * Enhance the theme types to include new properties from the CssVarsProvider.
+ * The theme is typed with CSS variables in `styled`, `sx`, `useTheme`, etc.
+ */
+declare module '@mui/material/styles' {
+  // The palette must be extended in each node.
+  interface Theme extends Omit<ColorSystem, 'palette'> {
+    prefix: string;
+    vars: CssVarsTheme['vars'];
+    getCssVar: CssVarsTheme['getCssVar'];
+    getColorSchemeSelector: CssVarsTheme['getColorSchemeSelector'];
+  }
+
+  interface Palette {
+    dividerChannel: string;
+    AppBar: PaletteAppBar;
```

Comment:
Let's add a comment that this needs to be sync with the changes in the other files. Same for the other type defined here.